### ISSUE TITLE
CSScript with localization resources

### ISSUE THREAD

**Issue by mspinder** [oleg-shilo/cs-script#177]
In a hosted environment, after I compile a script all resource lookups are significantly longer.  To reproduce:
1.  Create a form.  (I tested this using 4.6.1)
2.  Enable localization (this is just to create a .resx file to be compiled into the assembly).
3.  Add a resource string to the **project's** resources.
4.  Run the following code in the form constructor
```csharp
var time = DateTime.Now;
for (int i = 0; i < 1000; i++)
{
    ComponentResourceManager resources = new ComponentResourceManager(typeof(FormWithResources));
    resources.GetString("TestResourceString");
}
var delta1 = DateTime.Now - time;

CSScript.LoadCode(@"
    namespace TestNamespace
    {
        public class Test
        {
            public Test() {}
        }
    }");

time = DateTime.Now;
for (int i = 0; i < 1000; i++)
{
    ComponentResourceManager resources = new ComponentResourceManager(typeof(FormWithResources));
    resources.GetString("TestResourceString");
}
var delta2 = DateTime.Now - time;

MessageBox.Show($"Before: {delta1}\nAfter: {delta2}");
```

Notice the large time difference.

I was able to narrow this down to something in .NET's internal assembly.cs:InternalGetSatelliteAssembly().

Is there something is CSScript that might prevent this scenario?  Unfortunately, all localized designer generated code creates a ComponentResourceManager for control initialization.  And my application dynamically creates controls during runtime.  (I am working on minimizing that as a workaround for now).

Thanks

**Comment by oleg-shilo** [comment]
Thank you for providing such detailed test case. Still... Would you be able to share the actual VS project sample?

**Comment by mspinder** [comment]
Here is the project.  Let me know if there is anything else I can help with.

Thanks

[TestCSScriptResourceSpeed.zip](https://github.com/oleg-shilo/cs-script/files/3717838/TestCSScriptResourceSpeed.zip)

**Comment by oleg-shilo** [comment]
Found the cause and the work around!

I remembered that AppDomain object has this weird behavior that it always fires `AssemblyResolve` event for "non existing resources assembly" when it tries to load any resources as requested by resource manager. Even though user does not handle this event the resources are always loaded OK. Basically AppDomain invokes custom assembly probing (`AssemblyResolve`) for no good reason.

CS-Script does handle `AssemblyResolve` because it needs to probe for script dependencies. Thus the resource manager triggered `AssemblyResolve` events are also attempted to be resolved by CS-Script. While this is no issue for CS-Script - it simply cannot find the resources assembly and returns null, it does take some CPU time to handle the event.

Thus if it is important for you to avoid even a slight performance hit then you will need to disable CS-Script assembly probing (handling `AssemblyResolve` event) and either do it by yourself or completely stop doing asm probing if your business logic allows that.


Thus your sample will work fine if you upf\date it like that:

```C#
CSScript.LoadCode(@"
    namespace TestNamespace
    {
        public class Test
        {
            public Test() {}
        }
    }");

// insert this line
CSScript.AssemblyResolvingEnabled = false;

time = DateTime.Now;
```
